vignettes/debugging.Rmd

```diff
@@ -1,149 +1,205 @@
 
-Here, we are setting our log file to be a temporary file that will be cleaned up at the end
-of the R session. Here, the file extension is `.log`, but the file extension does not matter.
+Before we move on to specific scenarios, lets look at a case where we record multiple requests:
 
-```r
-vcr::vcr_configure(
-  dir = vcr::vcr_test_path("fixtures"),
-  log = TRUE,
-  log_opts = list(file = file.path(tempdir(), "vcr.log"))
-)
-```
+```{r}
+local({
+  local_vcr_configure_log()
+  local_cassette("debugging-2")
 
-With `log=TRUE` you can continue with debugging. Open the log file you set in a text
-editor or other location; or examine in your shell/terminal.
+  resp1 <- req_perform(request(httpbin$url("/get?x=1")))
+  resp2 <- req_perform(request(httpbin$url("/get?x=2")))
+  resp3 <- req_perform(request(httpbin$url("/get?x=3")))
+})
+```
 
-As an example, after running the block above
+Now if we deliberately change the url, you can see that it looks at all three previously saved requests before giving up.
 
-```r
-foo <- function() crul::ok('https://httpbin.org/get')
+```{r}
+#| error: true
+local({
+  local_vcr_configure_log()
+  local_cassette("debugging-2")
 
-test_that("foo works", {
-  vcr::use_cassette("testing", {
-    x <- foo()
-  })
-  expect_true(x)
+  resp4 <- req_perform(request(httpbin$url("/get?x=4")))
 })
 ```
 
-If we open the log file we'll see the logs for each step vcr takes in handling an HTTP request.
-The logs have information on what cassette was used, what exact time it was recorded, what 
-matchers were in use, the cassette options, and how a request is handled.
+You'll notice that this is substantially more complicated to understand, and you can imagine it only gets harder the more requests you have.
+That's a good reason to keep your tests small and simple, and limited to only a couple of requests.
 
-```
-[Cassette: 'testing'] - 2020-11-24 16:05:17 - Init. HTTPInteractionList w/ request matchers [method, uri] & 0 interaction(s): {  }
-[Cassette: 'testing'] - 2020-11-24 16:05:17 - Initialized with options: {name: testing, record: once, serialize_with: yaml, persist_with: FileSystem, match_requests_on: c("method", "uri"), update_content_length_header: FALSE, allow_playback_repeats: FALSE, preserve_exact_body_bytes: FALSE}
-[Cassette: 'testing'] - 2020-11-24 16:05:17 - Handling request: head https://httpbin.org/get (disabled: FALSE)
-[Cassette: 'testing'] - 2020-11-24 16:05:17 - Identified request type: (recordable) for head https://httpbin.org/get
-[Cassette: 'testing'] - 2020-11-24 16:05:17 -    Recorded HTTP interaction: head https://httpbin.org/get => 200 
-```
+## Common problems and theirs solutions
+
+Now that you understand how to use logging to see exactly what vcr is doing, we can talk about some solutions to common problems.
+
+### Re-used cassette name
+
+You need to make sure that every test has it's own unique cassette name, otherwise you won't find the responses that you're expecting.
```

Comment:
```suggestion
You need to make sure that every test has its own unique cassette name, otherwise you won't find the responses that you're expecting.
```